api/v1/controllers/samples.js

```diff
@@ -422,17 +416,9 @@ module.exports = {
          * channel
          */
         sampleModel.bulkUpsertByName(value, user, readOnlyFields)
-        .then((samples) => {
-          samples.forEach((sample) => {
-            if (!sample.isFailed) {
-              if (featureToggles.isFeatureEnabled('publishPartialSample')) {
-                publisher.publishPartialSample(sample);
-              } else {
-                publisher.publishSample(sample, subHelper.model);
-              }
-            }
-          });
-        });
+        .then((samples) =>
+          samples.filter((s) => !s.isFailed)
+          .forEach((s) => publisher.publishSample(s, subHelper.model)));
```

Comment:
We could potentially have problems with this when the number of samples from `bulkupsert` is a lot. I think here, we are iterating over the array twice and creating also a copy of it.